applications/SwimmingDEMApplication/python_scripts/swimming_DEM_analysis.py

```diff
@@ -609,175 +603,82 @@ def MakeAnalyticsMeasurements(self):
-            self.time = self.fluid_solution._GetSolver().AdvanceInTime(self.time)
-            self.step += 1
-            self.TellTime(self.time)
-
-            if coupling_scheme_type == "UpdatedDEM":
-                time_final_DEM_substepping = self.time + self.Dt
-
-            else:
-                time_final_DEM_substepping = self.time
-
-            #self.PerformEmbeddedOperations() TO-DO: it's crashing
-
-            self.UpdateALEMeshMovement(self.time)
-
-            # solving the fluid part
-            if self.step >= self.GetFirstStepForFluidComputation():
-                self.FluidSolve(
-                    self.time,
-                    solve_system=self.fluid_solve_counter.Tick() and not self.stationarity
-                    )
-
-            # assessing stationarity
-                if self.stationarity_counter.Tick():
-                    self.AssessStationarity()
-
-            # printing if required
-
-            if self.particles_results_counter.Tick():
-                self.io_tools.PrintParticlesResults(
-                    self.pp.variables_to_print_in_file,
-                    self.time,
-                    self.spheres_model_part)
+    def InitializeSolutionStep(self):
+        self.TellTime(self.time)
+        self.PerformInitialDEMStepOperations(self.time)
+        self.disperse_phase_solution.InitializeSolutionStep()
+        if self._GetSolver().CannotIgnoreFluidNow():
+            self.fluid_solution.InitializeSolutionStep()
+        super(SwimmingDEMAnalysis, self).InitializeSolutionStep()
 
-                self.PrintDrag(
-                    self.drag_list,
-                    self.drag_file_output_list,
-                    self.fluid_model_part,
-                    self.time)
+    def FinalizeSolutionStep(self):
+        # printing if required
+        if self._GetSolver().CannotIgnoreFluidNow():
+            self.fluid_solution.FinalizeSolutionStep()
 
-            if self.print_counter_updated_DEM.Tick():
+        self.disperse_phase_solution.FinalizeSolutionStep()
 
-                if coupling_level_type:
-                    self.projection_module.ComputePostProcessResults(self.spheres_model_part.ProcessInfo)
+        # applying DEM-to-fluid coupling
 
-                self.post_utils.Writeresults(self.time_dem)
+        if self.DEM_to_fluid_counter.Tick() and self.time >= interaction_start_time:
+            self._GetSolver().projection_module.ProjectFromParticles()
 
-            # solving the DEM part
+        # coupling checks (debugging)
+        if self.debug_info_counter.Tick():
+            self.dem_volume_tool.UpdateDataAndPrint(
+                self.pp.CFD_DEM["fluid_domain_volume"].GetDouble())
 
-            self.derivative_recovery_counter.Activate(self.time > interaction_start_time)
+        super(SwimmingDEMAnalysis, self).FinalizeSolutionStep()
 
-            if self.derivative_recovery_counter.Tick():
-                self.RecoverDerivatives()
+    def RunSolutionLoop(self):
```

Comment:
I think, this method is defined twice

Reply:
true... I just checked and they are identical definitions.